### ISSUE TITLE
Integrate Federated Sharing into the Calendar App

### ISSUE THREAD

**Issue by schiessle** [owncloud/calendar#357]
Since ownCloud 8.1 we have the concept of federated sharing for files. You can share a file with any user on any server by sharing it with `<user>@<ownCloudServer>`. We can even auto-complete this user IDs from the contacts app and from system wide address books synced between ownClouds.

I wonder if this wouldn't be a nice feature for the calendar app? Imagine if I could share a calender with any user at any ownCloud with the same federated cloud ID I also use to share files.

I'm not a CalDAV expert but this is how I think it could work:
- A ownCloud specific API would be used to send a shared calendar to a remote ownCloud server
- The remote server would basically receive a CalDAV URL and a auth token
- The remote server would then integrate the external CalDAV calendar by standard CalDAV techniques.

Opinions?

**Comment by georgehrke** [comment]
Please reopen in https://github.com/owncloud/core/issues :)
This is definitely related to the CalDAV server, which is in core, and to the GUI.

AFAIK CalDAV sharing has it's own mechanism for sharing among different CalDAV servers.

**Comment by schiessle** [comment]
@georgehrke my idea was that for sharing we would use the CalDAV sharing, as suggested by you. But the calendar app would need at lest the possibility to enter such federated cloud ids to the share dialog. Maybe you are right that this is already enough and handing over the federated share id to core would already be enough to create a share by CalDAV. But I think at least some logic is needed in the calendar app.

**Comment by georgehrke** [comment]
Yes, it will definitely take some GUI code, but please open an issue in core anyway. That's where most of the work has to be done :)

**Comment by schiessle** [comment]
@georgehrke sure, I will also open a issue in core and link it together with this issue. Thanks for re-open it!
